src/server/hll_family.cc

```diff
@@ -75,26 +75,30 @@ OpResult<int> AddToHll(const OpArgs& op_args, string_view key, CmdArgList values
   RETURN_ON_BAD_STATUS(op_res);
   auto& res = *op_res;
   if (res.is_new) {
-    hll.resize(getDenseHllSize());
-    createDenseHll(StringToHllPtr(hll));
+    hll.resize(getSparseHllInitSize());
+    initSparseHll(StringToHllPtr(hll));
   } else if (res.it->second.ObjType() != OBJ_STRING) {
     return OpStatus::WRONG_TYPE;
   } else {
     res.it->second.GetString(&hll);
```

Comment:
If I understood correctly, GetString() returns a copy of data rather than reference, like robj in Redis.
This would cause some overhead to copy from and paste to string `hll`.
Since string cannot be used in C, more overhead to copy `hll_sds` from and paste to string `hll`.

Reply:
Indeed. Please see my comment below.